earthfile2llb/withdockerrun.go

```diff
@@ -287,7 +287,7 @@ func (wdr *withDockerRun) load(ctx context.Context, opt DockerLoadOpt) error {
 	if err != nil {
 		return errors.Wrapf(err, "parse target %s", opt.Target)
 	}
-	mts, err := wdr.c.buildTarget(ctx, depTarget.String(), opt.Platform, opt.BuildArgs, false, false)
+	mts, err := wdr.c.buildTarget(ctx, depTarget.String(), opt.Platform, true, opt.BuildArgs, false, false)
```

Comment:
This doesn't seem like an obvious choice. For example, if putting together a set of images from a bunch of remote repos, usually you don't need the privileged mode just for building the image. I would make this required for the user to specify. e.g. `WITH DOCKER --allow-privileged --load=+target`.

Reply:
I was thinking that the only way this code would be hit would be after it went through https://github.com/earthly/earthly/blob/main/earthfile2llb/interpreter.go#L431-L433 which would fail if privileged wasn't available.

Now that you mention it, this does seem wrong. What about a case where there's two different remote targets to load, but you only want to give one privileged access? e.g.

```
WITH DOCKER --load=(github.com/earthly/foo+target --allow-privileged ) --load=github.com/other/repo+bar
```

maybe we need to add a allowPrivileged bool to DockerLoadOpt too?

Reply:
This would overlap with the new style build args unfortunately.

I wouldn't worry about it, because there are workarounds if the user wants finer granularity:

```
foo-target:
  FROM --allow-privileged github.com/earthly/foo+target

...

WITH DOCKER \
        --load=+foo-target \
        --load=github.com/other/repo+bar
```

Or

```
IMPORT --allow-privileged github.com/earthly/foo

...

WITH DOCKER \
        --load=foo+target \
        --load=github.com/other/repo+bar
```

Reply:
Those are very reasonable workaround, and good catch on this, I've added an AllowPrivileged bool to the DockerLoadOpt struct now.